mapie/quantile_regression.py

```diff
@@ -0,0 +1,486 @@
+        self.estimators_: List[RegressorMixin] = []
+
+        # Work
+        y_calib_preds = np.full(
+            shape=(3, self.n_samples_calib),
+            fill_value=np.nan
+        )
+
+        if isinstance(checked_estimator, Pipeline):
+            estimator = checked_estimator[-1]
+        else:
+            estimator = checked_estimator
+        name_estimator = estimator.__class__.__name__
+        alpha_name = self.quantile_estimator_params[
+            name_estimator]["alpha_name"]
+        for i, alpha_ in enumerate(alpha):
+            cloned_estimator_ = clone(checked_estimator)
+            params = {alpha_name: alpha_}
+            if isinstance(checked_estimator, Pipeline):
+                cloned_estimator_[-1].set_params(**params)
+            else:
+                cloned_estimator_.set_params(**params)
+            self.estimators_.append(fit_estimator(
+                cloned_estimator_, X, y, sample_weight
+            ))
+            y_calib_preds[i] = self.estimators_[-1].predict(X_calib)
+
+        self.conformity_scores_ = np.full(
+                shape=(3, self.n_samples_calib),
+                fill_value=np.nan
+            )
+        self.conformity_scores_[0] = y_calib_preds[0]-y_calib
+        self.conformity_scores_[1] = y_calib-y_calib_preds[1]
+        self.conformity_scores_[2] = np.max(
+            [
+                self.conformity_scores_[0],
+                self.conformity_scores_[1]
+            ], axis=0
+        )
+        return self
+
+    def predict(  # type: ignore
+        self,
+        X: ArrayLike,
+        symmetry: Optional[bool] = True,
+    ) -> Union[NDArray, Tuple[NDArray, NDArray]]:
+        """
+        Predict target on new samples with confidence intervals.
+        Residuals from the training set and predictions from the model clones
+        are central to the computation.
+        Prediction Intervals for a given ``alpha`` are deduced from the
+        quantile regression at the alpha values: alpha/2, 1 - (alpha/2)
+        while adding a constant based uppon their residuals.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            Test data.
+        symmetry : Optional[bool], optional
+            Deciding factor to whether to find the quantile value for
+            each residuals separatly or to use the maximum of the two
+            combined. This results in having either symmetric constants
+            added for both upper and lower bounds or not.
+
+        Returns
+        -------
+        Union[NDArray, Tuple[NDArray, NDArray]]
+
+        - NDArray of shape (n_samples,) if alpha is None.
+
+        - Tuple[NDArray, NDArray] of shapes
+        (n_samples,) and (n_samples, 2, n_alpha) if alpha is not None.
+
+            - [:, 0, :]: Lower bound of the prediction interval.
+            - [:, 1, :]: Upper bound of the prediction interval.
+        """
+        check_is_fitted(self, self.fit_attributes)
```

Comment:
Add something like:
```
alpha = self.alpha if symmetry else self.alpha / 2
```
in order to call once `check_alpha_and_n_samples` and `q` ?

Reply:
done